subprojects/base-services/src/main/java/org/gradle/internal/hash/Hashing.java

```diff
@@ -199,61 +198,41 @@ protected MessageDigest createDigest() {
     }
 
     private static class MessageDigestHasher implements PrimitiveHasher {
-        private final MessageDigest digest;
-        private final ByteBuffer buffer = ByteBuffer.allocate(8).order(ByteOrder.LITTLE_ENDIAN);
-        private boolean done;
+        private MessageDigest digest;
 
         public MessageDigestHasher(MessageDigest digest) {
             this.digest = digest;
         }
 
-        private void checkNotDone() {
-            if (done) {
-                throw new IllegalStateException("Cannot reuse hasher");
-            }
+        private MessageDigest getDigest() {
+            return Preconditions.checkNotNull(digest, "Cannot reuse hasher!");
```

Comment:
Nice!

Reply:
Actually, this should be:

```suggestion
            return Preconditions.checkState(digest != null, "Cannot reuse hasher!");
```

...so that we return an `InvalidStateException`.

Reply:
that can't return the digest, but I'll make it an ISE explicitly.
see: https://github.com/gradle/gradle/pull/9315#discussion_r279722956